funcs/func_groupcount.c

```diff
@@ -273,21 +1169,229 @@ static int group_list_function_read(struct ast_channel *chan, const char *cmd,
 	return 0;
 }
 
+static int group_match_list_function_read(struct ast_channel *chan, const char *cmd,
+				    char *data, char *buf, size_t len)
+{
+	struct ast_group_meta *gmi = NULL;
+	struct ast_str *foundgroup_str = ast_str_create(MAX_GROUP_LEN + MAX_CATEGORY_LEN);
+
+	buf[0] = '\0';
+
+	if (!foundgroup_str) {
+		ast_log(LOG_ERROR, "Unable to allocate new variable.  Cannot proceed with GROUP_MATCH_LIST()\n");
+		return -1;
+	}
+
+	char groupmatch[MAX_GROUP_LEN] = "";
```

Comment:
This code *still* does not compile properly... *please* make sure it compiles in developer mode before pushing for review.

```
func_groupcount.c: In function 'group_match_list_function_read':
func_groupcount.c:1185:9: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
 1185 |         char groupmatch[MAX_GROUP_LEN] = "";
```

If you can fix this quickly, that'd be appreciated, that way I don't need to write a temporary diff to fix compilation.

Reply:
Fixed!

Reply:
It still doesn't seem to be fixed when I look at the latest version.

Reply:
> It still doesn't seem to be fixed when I look at the latest version.

@gtjoseph Also, I think this came up before, but I remember there were previous changes the CI didn't catch this same compiler issue with (mixed declarations and code), and then it caused problems on when it was merged and compiled out in the wild.

The strange thing is it's actually causing compilation failures with my GitHub CI, so I'm not sure why the Asterisk one isn't getting tripped. Perhaps it's the gcc version? Just wanted to get your thoughts.